src/ol/PluggableMap.js

```diff
@@ -1139,11 +1111,24 @@ class PluggableMap extends BaseObject {
       targetElement = this.getTargetElement();
     }
 
-    if (this.keyHandlerKeys_) {
+    if (event.oldValue) {
       for (let i = 0, ii = this.keyHandlerKeys_.length; i < ii; ++i) {
         unlistenByKey(this.keyHandlerKeys_[i]);
       }
       this.keyHandlerKeys_ = null;
+      this.viewport_.removeEventListener(
+        EventType.CONTEXTMENU,
+        this.boundHandleBrowserEvent_
+      );
+      this.viewport_.removeEventListener(
+        EventType.WHEEL,
+        this.boundHandleBrowserEvent_
+      );
+      if (this.handleResize_ !== undefined) {
+        removeEventListener(EventType.RESIZE, this.handleResize_, false);
+        this.handleResize_ = undefined;
```

Comment:
Not sure if we can remove the same logic in the `(!targetElement)` branch below? I couldn't think of a case where it would still trigger. (Only 2 subsequent `setTarget(null)` but then there wouldn't be an `this.handleResize_`)

Reply:
Good catch, thanks! That was a copy/paste mistake.